apps/portal/controllers/apis/rt.jag

```diff
@@ -93,97 +110,97 @@
-                    name: "TIMESTAMP",
-                    type: "time"
+                    name: metaData[i].getAttributeName(),
+                    type: metaData[i].getAttributeType()
                 });
-
-                var allDataLength = metaDataLength + correlationDataLength + payloadDataLength;
-                for (var i = 0; i < metaDataLength; i++) {
-                    output.push({
-                        name: metaData[i].getAttributeName(),
-                        type: metaData[i].getAttributeType()
-                    });
-                    counter++;
-                }
-                for (var i = 0; i < correlationDataLength; i++) {
-                    output.push({
-                        name: correlationData[i].getAttributeName(),
-                        type: correlationData[i].getAttributeType()
-                    });
-                    counter++;
-                }
-                for (var i = 0; i < payloadDataLength; i++) {
-                    output.push({
-                        name: payloadData[i].getAttributeName(),
-                        type: payloadData[i].getAttributeType()
-                    });
-                    counter++;
-                }
-            } catch (e) {
-                //If Unauthorized Java exception occurred.
-                //User will be logged out and Unauthorized 401 HTTP code will be sent
-                if(e.javaException instanceof org.apache.axis2.AxisFault){
-                    if (e.message.split(":")[3].trim() == "Unauthorized") {
-                        log.error(e.message);
-                        var usr = require('/modules/user.js');
-                        usr.logout();
-                        response.sendError(401);
-                    }
-                }
+                counter++;
+            }
+            for (var i = 0; i < correlationDataLength; i++) {
+                output.push({
+                    name: correlationData[i].getAttributeName(),
+                    type: correlationData[i].getAttributeType()
+                });
+                counter++;
             }
-        } else {
-            var schema = analyticsStub.getTableSchema(tableName);
-            var columns = schema.getColumns();
-            for (var i = 0; i < columns.length; i++) {
-                var column = columns[i];
+            for (var i = 0; i < payloadDataLength; i++) {
                 output.push({
-                    name : column.getColumnName(),
-                    type : column.getColumnType()
+                    name: payloadData[i].getAttributeName(),
+                    type: payloadData[i].getAttributeType()
                 });
-            };
+                counter++;
+            }
+        } catch (e) {
+            //If Unauthorized Java exception occurred.
+            //User will be logged out and Unauthorized 401 HTTP code will be sent
+            if (e.javaException instanceof org.apache.axis2.AxisFault) {
```

Comment:
why are we only handling the "AxisFault" error here? what's with other errors and exceptions ?